adapters/kagome/src/host_api/child_storage.cpp

```diff
@@ -70,14 +67,82 @@ namespace child_storage {
   }
 
   void read_version_1(helpers::RuntimeEnvironment environment,
-                      const std::vector<std::string> &args) {
+                      const std::vector<std::string> &inputs) {
     throw NotImplemented();  // TODO not implemented
+    // Parse inputs
+    BOOST_ASSERT(inputs.size() == 6);
+
+    const std::string_view child1 = inputs[0];
+    const std::string_view child2 = inputs[1];
+    const std::string_view key = inputs[2];
+    const std::string_view value = inputs[3];
+    const uint32_t offset = std::stoul(inputs[4]);
+    const uint32_t length = std::stoul(inputs[5]);
+
+    // Check that key has not been set
+    auto result = environment.execute<helpers::MaybeBuffer>(
+        "rtm_ext_default_child_storage_read_version_1", child1, key, offset, length);
+
+    BOOST_ASSERT_MSG(!result, "Child1 data exists");
+
+    // Add data to storage
+    environment.execute<void>(
+        "rtm_ext_default_child_storage_set_version_1", child1, key, value);
+
+    // Get invalid key (different child storage)
+    result = environment.execute<helpers::MaybeBuffer>(
+        "rtm_ext_default_child_storage_read_version_1", child2, key, offset, length);
+
+    BOOST_ASSERT_MSG(result.has_value(), "Child2 no value");
+
+    // Get valid key
+    result = environment.execute<helpers::MaybeBuffer>(
+        "rtm_ext_default_child_storage_read_version_1", child1, key, offset, length);
+
```

Comment:
Maybe add an assert for `result.has_value()` here.

Reply:
Ok